src/plugins/workspace/server/utils.ts

```diff
@@ -113,3 +115,25 @@ export const getDataSourcesList = (client: SavedObjectsClientContract, workspace
       }
     });
 };
+
+export const checkAndSetDefaultDataSource = async (
+  uiSettingsClient: IUiSettingsClient,
+  dataSources: string[],
+  isNeededCheck: boolean
```

Comment:
```suggestion
  needCheck: boolean
```
Nit: `isNeededCheck` seems a little bit too much for me.

Reply:
Updated.